src/main/java/com/xebialabs/deployit/community/argos/Argos4jVerifier.java

```diff
@@ -0,0 +1,171 @@
+        NonPersonalAccount npaAccount = version.getApplication().getProperty(ArgosConfiguration.PROPERTY_ARGOS_PERSONAL_ACCOUNT);
+        if (npaAccount == null) {
+            context.logError(String.format("Argos NPA not set on Application [%s]", version.getApplication().getName()));
+            return false;
+        }
+        if (supplyChain == null) {
+            context.logError(String.format("Argos Supply Chain not set on Application [%s]", version.getApplication().getName()));
+            return false;
+        }
+        char[] passphrase = npaAccount.getPassphrase().toCharArray(); 
+        
+        List<String> path = getPath(supplyChain);
+        String supplyChainName = getSupplyChainName(supplyChain);
+        
+        Argos4jSettings settings = Argos4jSettings.builder()
+                .pathToLabelRoot(path)
+                .supplyChainName(supplyChainName)
+                .signingKeyId(npaAccount.getKeyId())
+                .argosServerBaseUrl(ArgosConfiguration.getArgosServerBaseUrl()).build();
+        Argos4j argos4j = new Argos4j(settings);
+        
+        VerifyBuilder verifyBuilder = argos4j.getVerifyBuilder();
+        
+        verifyBuilder.addFileCollector(getDarCollector(context, version));
+        
+        Set<FileCollector> fileCollectors = getRemoteFileCollectors(context, version);
+        
+        fileCollectors.forEach(verifyBuilder::addFileCollector);
+        
+        VerificationResult verifyResult = null;
+        try {
+            verifyResult = verifyBuilder.verify(passphrase);
+        } catch (Argos4jError exc) {
+            context.logError(String.format("Exception during Argos Notary verify: [%s]", exc.getMessage()));
+            return false;
+        }
+        if (verifyResult != null && verifyResult.isRunIsValid()) {
+            context.logOutput(String.format("Application [%s] version [%s] is valid according to Argos Notary", version.getApplication().getName(), version.getName()));
+            return true;
+        } else {
+            context.logError(String.format("Application [%s] version [%s] is invalid according to Argos Notary", version.getApplication().getName(), version.getName()));
+            return false;
+        }
+    }
+    
+    private static RemoteZipFileCollector getDarCollector(ExecutionContext context, Version version) {
+        XldClientConfig xldConf = ArgosConfiguration.getXldClientConfig(context);
+        String downloadKey = getVersionDownloadKey(xldConf, version.getId());
+        return RemoteZipFileCollector.builder()
+                .url(ArgosConfiguration.getXldUrlForExport(downloadKey))
+                .username(xldConf.getUsername()).password(xldConf.getPassword().toCharArray())
+                .build();
+
+    }
+    
+    private static Set<FileCollector> getRemoteFileCollectors(ExecutionContext context, Version version) {
```

Comment:
list is denk ik handiger omdat FileCollector geen equals en hashcode functie heeft

Reply:
er wordt altijd gewaarschuwd over het niet implementeren van equals en hashcode. Maar ik heb het aangepast